components/tidb_query/src/batch/runner.rs

```diff
@@ -362,6 +371,9 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
                     self.out_most_executor.schema().len()
                 );
                 let mut chunk = Chunk::default();
+                if !is_arrow_encodable(self.out_most_executor.schema()) {
```

Comment:
Prefer to put it in `from_request`, so that you don't need to check it every time.

Reply:
Additionally L383 can be removed (it is not introduced in this PR, however you can do it conveniently).